content/en/docs/v3.6/learning/api_guarantees.md

```diff
@@ -1,75 +1,60 @@
+etcd is a consistent and durable key value store.
+The key value store is exposed through the KV and Watch [gRPC Services].
+etcd ensures the strongest consistency and durability guarantees for a distributed system.
+This specification enumerates the API guarantees made by etcd.
 
-### APIs to consider
+### RPCs to consider
 
-* Read APIs
-    * range
-    * watch
-* Write APIs
-    * put
-    * delete
-* Combination (read-modify-write) APIs
-    * txn
-* Lease APIs
-    * grant
-    * revoke
-    * put (attaching a lease to a key)
+* KV Service
+  * [Range](../api/#range)
+  * [Put](../api/#put)
+  * [Delete Range](../api/#delete-range)
+  * [Transaction](../api/#transaction)
+* Watch Service
+  * [Watch](../api/#watch-api)
+* Lease Service
+  * [Lease grant](../api/#obtaining-leases)
+  * [Keep alive](../api/#keep-alives)
 
-### etcd specific definitions
+KV Service allows for direct reading and manipulation of key value store.
+Watch Service allows subscribing to key value store changes.
+Lease Service allows assigning a time to live to a key.
 
-#### Operation completed
+KV Service RPCs will take an immediate effect, while Watch Service RPC will return with some unbounded delay.
+In correctly working etcd cluster you should expect to see watch events to appear with 10ms delay after them happening.
+However, there is no limit and events in unhealthy clusters might never arrive.
 
-An etcd operation is considered complete when it is committed through consensus,
-and therefore “executed” -- permanently stored -- by the etcd storage engine.
-The client knows an operation is completed when it receives a response from the
-etcd server. Note that the client may be uncertain about the status of an
-operation if it times out, or there is a network disruption between the client
-and the etcd member. etcd may also abort operations when there is a leader
-election. etcd does not send `abort` responses to  clients’ outstanding requests
-in this event.
+## KV Service
 
-#### Revision
+etcd ensures durability and strict serializability for all RPCs under KV Service.
+Those are the strongest isolation guarantee of distributed transactional database systems.
 
-An etcd operation that modifies the key value store is assigned a single
-increasing revision. A transaction operation might modify the key value store
-multiple times, but only one revision is assigned. The revision attribute of a
-key value pair that was modified by the operation has the same value as the
-revision of the operation. The revision can be used as a logical clock for key
-value store. A key value pair that has a larger revision is modified after a key
-value pair with a smaller revision. Two key value pairs that have the same
-revision are modified by an operation "concurrently".
+### Durability
+
+Any completed operations are durable. All accessible data is also durable data.
+A read will never return data that has not been made durable.
+
+### Strict serializability
```

Comment:
ACID are four fundamental attributes of transactional database. No matter what consistency modes (e.g. linearizability, serializablity, etc.) we are describing here, they are just a type of consistency mode. Please consider to revert to original structure.

Reply:
FYI etcd is not transactional database. It's key value store, that's why we don't invoke full acid here.

Reply:
> FYI etcd is not transactional database. It's key value store, that's why we don't invoke ACID here.

etcd is a K/V store, and it also supports transaction, and supports the four fundamental attributes ACID.

> Please explain what you mean by "original structure".

https://etcd.io/docs/v3.5/learning/api_guarantees/#guarantees-provided

Reply:
Those are conditional statements not real transactions as you cannot revert them.  Please note that ACID was never mentioned in the document. If you have suggestion about structure let's discuss them. For me problem with the old document was lack of structure. 

Old document "structure":
* atomicity
* durability
* Isolation level and consistency
  * strict serializability
  * linearizability
  
Problems with it:
* strict serializability implies atomicity and linearizability.
* Isolation level is not a guarantee.

Proposed structure
* Durability
* Strict serializability, which implies
  * Atomicity
   * Linearizability
   
If you think this is not an improvement, please propose some structure. Note I left atomicity and linearizability to provide more granularity when explaining strict serializability. I could remove them if you prefer.

Reply:
Indeed Strict serializability implies other items, but the concept of ACID has already been widely accepted. I do think etcd supports ACID although it isn't clearly documented.

I won't insist on this.

Reply:
Yea, I noticed a problem with two independent models for naming consistency guarantee. ACID coming from SQL databases, and [consistency models](https://jepsen.io/consistency) coming from distributed systems.

I don't claim to know which is better for etcd documentation, so I wanted to leave the previously present model borrowed from jepsen.